changelog.d/12772.bugfix

```diff
@@ -0,0 +1 @@
+Moves methods that call `add_push_rule` to the `PushRuleStore` class.
```

Comment:
I wouldn't say this was a bug since its not user visible, so I think it should be a `.misc` entry.